app/src/main/java/im/status/applet_installer_test/appletinstaller/APDUResponse.java

```diff
@@ -35,6 +37,14 @@ public boolean isOK() {
         return this.sw == SW_OK;
     }
 
+    public APDUResponse checkOK() throws IOException {
+        if (!isOK()) {
+            throw new IOException("Unexpected error SW");
```

Comment:
I think you can use `APDUException` passing `this.getSW()` and `"unexpected SW"`